components/permalink_view/actions.ts

```diff
@@ -19,12 +23,13 @@ import {joinPrivateChannelPrompt} from 'utils/channel_utils';
 import {ActionTypes, Constants, ErrorPageTypes} from 'utils/constants';
 import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';
 import {isComment, getPostURL} from 'utils/post_utils';
+import {GlobalState} from 'types/store';
 
 let privateChannelJoinPromptVisible = false;
 
-function focusRootPost(post, channel) {
-    return async (dispatch, getState) => {
-        const postURL = getPostURL(getState(), post);
+function focusRootPost(post: Post, channel: Channel) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const postURL = getPostURL(getState() as GlobalState, post);
```

Comment:
I am casting it as `GlobalState` because GlobalState from `types/store` and GlobalState from `@mattermost/types/store` are different. 

`getState()` is returning GlobalState from `@mattermost/types/store` but `getPostURL` is expecting GlobalState from `types/store` 

Please let me know if it is okay. Alternatively I can change type in `getPostURL from 'utils/post_utils';`